mne/html_templates/_templates.py

```diff
@@ -1,10 +1,105 @@
 # License: BSD-3-Clause
 # Copyright the MNE-Python contributors.
+import datetime
 import functools
+import uuid
+from collections import defaultdict
+from typing import Any, Literal, Union
+
+from .._fiff.pick import channel_type
+from ..defaults import _handle_default
 
 _COLLAPSED = False  # will override in doc build
 
 
+def _format_number(value: Union[int, float]) -> str:
+    """Insert thousand separators."""
+    return f"{value:,}"
+
+
+def _append_uuid(string: str, sep: str = "-") -> str:
+    """Append a UUID to a string."""
+    return f"{string}{sep}{uuid.uuid1()}"
+
+
+def _data_type(obj) -> str:
+    """Return the qualified name of a class."""
+    return obj.__class__.__qualname__
+
+
+def _dt_to_str(dt: datetime.datetime) -> str:
+    """Convert a datetime object to a human-reaable string representation."""
+    return dt.strftime("%B %d, %Y    %H:%M:%S") + " UTC"
```

Comment:
does `%Z` not work here?
https://docs.python.org/3/library/datetime.html#strftime-and-strptime-format-codes

Reply:
This could work, yes
I can try it out

Reply:
This seems to work!

While we're at it: WDYT about a date representation like this:

```
2002-12-03 at 19:01:10 UTC
```
(YYYY-MM-DD)

Reply:
> YYYY-MM-DD

That is my personal preferred date format, so I'm fine with it :)
It's also what ISO 8601 stipulates, so it's as good a choice as any

Reply:
@drammock I addressed this in [fd976a5](https://github.com/mne-tools/mne-python/pull/12583/commits/fd976a53c5ce2386f70dcc1664a52e3cb256c9ad)

I didn't implement "full" ISO-8601, because it's not super human-readable when you have a datetime string; but I took inspiration from that standard. Please LMK what you think